src/base/emu-i386/simx86/interp.c

```diff
@@ -2536,8 +2540,15 @@ intop3b:		{ int op = ArOpsFR[D_MO(opc)];
 			PC++; } break;
 /*ec*/	case INvb: {
 			unsigned short a;
+			unsigned char uc;
 			CODE_FLUSH();
 			a = rDX;
+			if ((CEmuStat & CeS_INSTREMU) &&
+			    (uc=VGA_emulate_inb(a)) != 0xff) {
```

Comment:
What is 0xff?

Reply:
this is just copied from the instremu code, so it returns 0xff for non-VGA ports but it's a bit dodgy indeed, as it relies on VGA ports never returning 0xff.

Reply:
I've changed it to be like getchar(), returning an int, where -1 means it's not a VGA port, and outb also returns -1 then.